tasks/installation.yml

```diff
@@ -1,33 +1,86 @@
 ---
 # tasks file for sap-preconfigure: installation
 
-- name: ensure minimal required packages are installed
+- name: Ensure required package groups are installed
   package:
     state: present
-    name: "{{ sap_preconfigure_min_pkgs }}"
-  when:
-    - sap_preconfigure_min_package_check|bool
-    - not( (sap_preconfigure_min_pkgs is undefined) or (sap_preconfigure_min_pkgs is none) or (sap_preconfigure_min_pkgs | trim == '') )
-
-- name: ensure required package groups are installed
-  package:
-    state: present
-    name: "{{ sap_preconfigure_packagegroups }}"
-  check_mode: "{{ sap_preconfigure_installation_check_only }}"
+    name: "{{ __sap_preconfigure_packagegroups }}"
 
-- name: ensure required packages are installed
+- name: Ensure required packages are installed
   package:
     state: present
-    name: "{{ sap_preconfigure_packages }}"
-  check_mode: "{{ sap_preconfigure_installation_check_only }}"
+    name: "{{ __sap_preconfigure_packages }}"
   register: sap_preconfigure_register_groupinstall
 
-- name: ensure system is updated to the latest patchlevel
+- name: Ensure minimum packages are installed
+  block:
+    - name: check if minimum release needs to be installed
+      shell: |
+             set -x
+             required_pkg={{ pkg | join('-') }}
+             newest=$(echo -e "$required_pkg\n$(rpm -q {{ pkg[0] }} )"| sort -V | tail -1)
+             if [ $newest == $required_pkg ]; then
+                echo $newest
+             fi
+      loop: "{{ __sap_preconfigure_min_pkgs }}"
+      loop_control:
+               loop_var: pkg
+      check_mode: no
+      register: __sap_preconfigure_register_minpkglist
+      changed_when: false
+
+    - name: Initialize an empty list for our strings
+      set_fact:
+        __sap_preconfigure_fact_minpkglist: []
+
+    - name: Create list of packages to be installed
+      set_fact:
+        __sap_preconfigure_fact_minpkglist: "{{ __sap_preconfigure_fact_minpkglist | difference(['']) + [ pkg.stdout ] }}"
+      loop: "{{ __sap_preconfigure_register_minpkglist.results }}"
+      loop_control:
+               loop_var: pkg
+    - debug: var=__sap_preconfigure_fact_minpkglist
+
+    - name: Install minimum packages if required
+      yum:
```

Comment:
do you need to use `yum` here instead of `package`?  Not sure if that will work on RHEL8.

Reply:
I agree that we should be linux distro independent as much as possible. I tested the behavior of package instead of yum on RHEL 8.1 and found that it works.
Fixed in https://github.com/berndfinger/sap-preconfigure/commit/4a1f8c15991a4c8369ab56f7c457c49a96e76bcb .